src/parse.jl

```diff
@@ -0,0 +1,254 @@
+                elseif nodename(field) == "transition"
+                    tcount[id_component] += 1
+                end
+            end
+        end
+    end
+    return (lcount, tcount)
+end
+
+"""
+    parse_sxmath(s)
+
+Returns the list of expressions corresponding to a given SX string.
+
+### Input
+
+- `s` - string
+
+### Output
+
+Vector of expressions, equations or inequalities.
+
+### Examples
+
+```jldoctest
+julia> import SX.parse_sxmath
+
+julia> parse_sxmath("x >= 0")
+1-element Array{Expr,1}:
+ :(x >= 0)
+
+julia> parse_sxmath("x' == x & v' == -0.75*v")
+2-element Array{Expr,1}:
+ :(x' = x)
+ :(v' = -0.75v)
+
+julia> parse_sxmath("x == 0 & v <= 0")
+2-element Array{Expr,1}:
+ :(x = 0)
+ :(v <= 0)
+```
+
+### Algorithm
+
+Consists of the following steps (in the given order):
+
+- split the string with the `&` key
+- remove trailing whitespace of each substring
+- replace double `==` with single `=`
+- cast to a Julia expression with `parse`
+"""
+parse_sxmath(s) = parse.(replace.(strip.(split(s, "&")), "==", "="))
+
+"""
+    parse_sxmodel!(root_sxmodel, HDict)
+
+### Input
+
+- `HDict` - dictionary that wraps the hybrid model and contains the keys `(automaton,
+            variables, transitionlabels, invariants, flows, resetmaps, switchings)`
+
+### Output
+
+The `HDict` dictionary.
+
+### Notes
+
+Edge labels are not used and their symbol is (arbitrarily) set to the integer 1.
+"""
+function parse_sxmodel!(root_sxmodel, HDict)
+
+    # edge labels
+    σ = 1
+
+    # counter for the variables
+    id_variable = 0
+
+    # counter for the transitions
+    id_transition = 0
+
+    for component in eachelement(root_sxmodel)
+        for field in eachelement(component)
+            if nodename(field) == "param"
+                if field["type"] == "real"
+                    id_variable += 1
+                    add_variable!(HDict["variables"], field, id_variable)
+                elseif field["type"] == "label"
+                    add_transition_label!(HDict["transitionlabels"], field)
+                else
+                    error("param type unknown")
+                end
+            elseif nodename(field) == "location"
+                (id_location, i, f) = parse_location(field)
+                HDict["invariants"][id_location] = i
+                HDict["flows"][id_location] = f
+            elseif nodename(field) == "transition"
+                id_transition += 1
+                (q, r, guard, assignment) = parse_transition(field)
+                add_transition!(HDict["automaton"], q, r, σ)
+                HDict["switchings"][id_transition] = guard
+                HDict["resetmaps"][id_transition] = assignment
+            else
+                error("node $(nodename(field)) unknown")
+            end
+        end
+    end
+    return HDict
+end
+
+"""
+    add_variable!(variables, field, id=1)
+
+### Input
+
+- `variables` -- vector of symbolic variables
+- `field`     -- node with a `param` variable field
```

Comment:
node?

Reply:
`- `field`     -- an `EzXML.Node` node with containing information about a `param` field`

Reply:
for example the object that contains this info would be a node:

```
<param name="x" type="real" d1="1" d2="1" local="false" dynamics="any" controlled="true" />
```

Reply:
EzXML can parse pieces of XML as strings very easily; i'm adding this to the unit tests:

```julia
using EzXML
@testset "Variables" begin
    X = parsexml("""
                 <param name="x" type="real" d1="1" d2="1" local="false" dynamics="any" controlled="true" />
                 """)
    vars = Dict()
    SX.add_variable!(vars, root(X))
    @test collect(keys(vars)) == [:x]
    @test vars[:x]["local"] == false
    @test vars[:x]["controlled"] == true
    @test vars[:x]["dynamics"] == "any"
    @test vars[:x]["id"] == 1
end
```